packages/apps/plugins/plugin-markdown/src/extensions.tsx

```diff
@@ -50,17 +51,13 @@ export const getExtensions = ({ viewMode, settings, document, query, dispatch }:
       : [
           decorateMarkdown({
             selectionChangeDelay: 100,
+            // TODO(wittjosiah): For internal links, consider ignoring the link text and rendering the label of the object being linked to.
             renderLinkButton:
               dispatch && document
                 ? onRenderLink((id: string) => {
                     void dispatch({
-                      action: NavigationAction.ADD_TO_ACTIVE,
-                      data: {
-                        id,
-                        part: 'main',
-                        pivot: { id: fullyQualifiedId(document) },
-                        scrollIntoView: true,
-                      },
+                      action: NavigationAction.OPEN,
+                      data: { activeParts: { main: [id] } },
```

Comment:
Maybe `add_to_active` should handle solo mode instead of switching to NavigationAction.OPEN? I like the idea of internal links opening as neighbours because links imply relatedness.

Reply:
How hard do you think it would be for add to active to support solo mode?

(I actually wondered if it was being deprecated because I didn't see it being used much at all)

Reply:
We use it in the search plugin to insert a plank under the `+` sign. I think it would be pretty easy to support solo. In the intent handler, if the layout mode is solo it should dispatch `OPEN` and return otherwise perform the current logic.

Reply:
okay, updated